pkg/rewards/show.go

```diff
@@ -85,186 +82,192 @@ func ShowRewards(cCtx *cli.Context) error {
 	}
 
-	if config.ClaimType == All {
-		requestBody := map[string]string{
-			"earnerAddress": config.EarnerAddress.String(),
-			"days":          fmt.Sprintf("%d", absInt64(config.NumberOfDays)),
-		}
-		resp, err := post(
-			fmt.Sprintf("%s/%s", url, GetEarnedTokensForStrategyEndpoint),
-			requestBody,
-		)
-		if err != nil {
-			return err
-		}
-		defer resp.Body.Close()
+	elReader, err := elcontracts.NewReaderFromConfig(
+		elcontracts.Config{
+			RewardsCoordinatorAddress: config.RewardsCoordinatorAddress,
+		},
+		ethClient, logger,
+	)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to create new reader from config", err)
+	}
+
+	df := httpProofDataFetcher.NewHttpProofDataFetcher(
+		config.ProofStoreBaseURL,
+		config.Environment,
+		config.Network,
+		http.DefaultClient,
+	)
 
-		var responseBody RewardResponse
-		err = json.NewDecoder(resp.Body).Decode(&responseBody)
+	claimDate, _, err := getClaimDistributionRoot(ctx, "latest_active", elReader, logger)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to get claim distribution root", err)
+	}
+
+	proofData, err := df.FetchClaimAmountsForDate(ctx, claimDate)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to fetch claim amounts for date", err)
+	}
+
+	tokenAddressesMap, present := proofData.Distribution.GetTokensForEarner(config.EarnerAddress)
+	if !present {
+		return eigenSdkUtils.WrapError("earner address not found in distribution", nil)
+	}
+
+	allRewards := make(map[gethcommon.Address]*big.Int)
+	for pair := tokenAddressesMap.Oldest(); pair != nil; pair = pair.Next() {
+		amt, _ := new(big.Int).SetString(pair.Value.String(), 10)
+		allRewards[pair.Key] = amt
+	}
+
+	if config.ClaimType == All {
+		fmt.Println(strings.Repeat("-", 30), "All Rewards", strings.Repeat("-", 30))
+		err := handleRewardsOutput(config.Output, config.OutputType, allRewards)
 		if err != nil {
 			return err
 		}
-		normalizedRewards := normalizeRewardResponse(responseBody)
-		if common.IsEmptyString(config.Output) {
-			printNormalizedRewardsAsTable(normalizedRewards)
-		} else {
-			logger.Debugf("Writing total rewards to %s", config.Output)
-			err = common.WriteToCSV(normalizedRewards, config.Output)
+	} else {
+		claimedRewards := make(map[gethcommon.Address]*big.Int)
+		for address, _ := range allRewards {
+			claimed, err := elReader.GetCumulativeClaimed(&bind.CallOpts{Context: ctx}, config.EarnerAddress, address)
 			if err != nil {
 				return err
 			}
-			logger.Infof("Total rewards written to %s", config.Output)
-		}
-	} else if config.ClaimType == Unclaimed {
-		requestBody := map[string]string{
-			"earnerAddress": config.EarnerAddress.String(),
+			if claimed == nil {
+				claimed = big.NewInt(0)
+			}
+			claimedRewards[address] = claimed
 		}
-		claimableRewardsUrl := fmt.Sprintf("%s/%s", url, GetClaimableRewardsEndpoint)
-		resp, err := post(claimableRewardsUrl, requestBody)
-		if err != nil {
-			return err
+		if config.ClaimType == Claimed {
```

Comment:
consider 1) making this a switch statement on config.ClaimType  2) breaking out the cases into separate functions to make this function smaller